url.go

```diff
@@ -49,6 +53,7 @@ var ErrUnresolvedUrl error = fmt.Errorf("entity url series is not resolved")
 var (
 	validSeries = regexp.MustCompile("^[a-z]+([a-z0-9]+)?$")
 	validName   = regexp.MustCompile("^[a-z][a-z0-9]*(-[a-z0-9]*[a-z][a-z0-9]*)*$")
+	validStore  = regexp.MustCompile(fmt.Sprintf("^https?://(www\\.)?%s/", charmstoreURL))
```

Comment:
I'd be tempted use url.Parse to parse
the URL, and then do the rest of the checking on the parts
of the parsed URL.